honeybee_radiance/lightsource/sky/climatebased.py

```diff
@@ -0,0 +1,310 @@
+            epw: A Ladybug EPW objects.
+            month: An intger between 1-12 for month.
+            day: An intger between 1 to 28-31 depending on the input month.
+            hour: A float number larger or equal to 0 and smaller than 24.
+            direct_normal_irradiance: Direct normal irradiance (W/m2).
+            diffuse_horizontal_irradiance: Diffuse horizontal irradiance (W/m2).
+            north_angle: North angle in degrees. A number between -360 and 360 for the
+                counterclockwise difference between the North and the positive Y-axis in
+                degrees. 90 is West and 270 is East (Default: 0).
+            ground_reflectance: Average ground reflectance (Default: 0.2).
+        """
+        assert isinstance(epw, EPW), \
+            'epw must be from type EPW not {}'.format(type(epw))
+        location = epw.location
+        hoy = int(DateTime(month, day, hour).hoy)
+        direct_normal_irradiance = epw.direct_normal_radiation[hoy]
+        diffuse_horizontal_irradiance = epw.diffuse_horizontal_radiation[hoy]
+
+        return cls.from_lat_long(
+            location.latitude, location.longitude, location.time_zone, month, day, hour,
+            direct_normal_irradiance, diffuse_horizontal_irradiance, north_angle,
+            ground_reflectance)
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create the sky from a dictionary.
+
+        Args:
+            input_dict: A python dictionary in the following format
+
+        .. code-block:: python
+
+                {
+                'altitude': 0.0,
+                'azimuth': 0.0,
+                'direct_normal_irradiance': 0,
+                'diffuse_horizontal_irradiance': 0,
+                'ground_reflectance': 0.2,
+                'ground_hemisphere': {},  # see ground.Ground class [optional],
+                'sky_hemisphere': {}  # see hemisphere.Hemisphere class [optional]
+                }
+        """
+        sky = cls(
+            input_dict['altitude'],
+            input_dict['azimuth'],
+            input_dict['direct_normal_irradiance'],
+            input_dict['diffuse_horizontal_irradiance'],
+            input_dict['ground_reflectance']
+        )
+
+        if 'ground_hemisphere' in input_dict:
+            sky._ground_hemisphere = Ground.from_dict(input_dict['ground_hemisphere'])
+
+        if 'sky_hemisphere' in input_dict:
+            sky._sky_hemisphere = Hemisphere.from_dict(input_dict['sky_hemisphere'])
+
+        return sky
+
+    # TODO: add support for additional parameters
+    # TODO: add gendaylit to radiance-command and use it for validating inputs
+    def to_radiance(self, output_type=0):
+        """Return Radiance description of the sky.
+
+        Args:
+            output_type: An integer between 0 to 2 for output type.
+                [0] output in W/m2/sr visible (default)
+                [1] output in W/m2/sr solar
+                [2[ outputin lm/m2/sr luminance
```

Comment:
Backwards `[`. Also, I don't know if this is really the best way to have it show up in the sphinx docs.
Santiago used this convention in his docs cleanup:
https://github.com/ladybug-tools/ladybug-comfort/blob/master/ladybug_comfort/parameter/adaptive.py#L41-L48

Reply:
Good to know. I was actually trying to follow Santiago's convention. This one that you linked will work and will look better. I will change this.